Sources/NIOCore/NIOLoopBound.swift

```diff
@@ -127,6 +127,28 @@ public final class NIOLoopBoundBox<Value>: @unchecked Sendable {
         .init(_value: value, uncheckedEventLoop: eventLoop)
     }
 
+    #if compiler(>=6.0)
+    /// Initialise a ``NIOLoopBoundBox`` by sending a  value, validly callable off `eventLoop`.
+    ///
+    /// Contrary to ``init(_:eventLoop:)``, this method can be called off `eventLoop` because `value` is moved into the box and can no longer be accessed outside the box.
+    /// So we don't need to protect `value` itself, we just need to protect the ``NIOLoopBoundBox`` against mutations which we do because the ``value``
+    /// accessors are checking that we're on `eventLoop`.
+    public static func makeBoxTakingValue(
```

Comment:
I think if we're going to add this API, it needs to have the same name as the one above. It's ultimately the same function, with some slightly less strong generic constraints.